tests/performance-tests/src/configuration/rampingRequestRateProfile.ts

```diff
@@ -0,0 +1,65 @@
+import { Options } from 'k6/options';
+import { mergeObjects, parseIntOptional } from '../utils/utils';
+
+interface Config {
+  // Name of the scenario.
+  scenario?: string;
+  // Total duration of the main stage of the test, providing a steady rate of requests.
+  mainStageDurationMinutes: number;
+  // The maximum requests per second to ramp up to during the main stage.
+  maxRequestRatePerSecond: number;
+  // Duration of a short final stage to allow any long-running requests to complete.
+  cooldownStageDurationMinutes: number;
+}
+
+const overrides: Partial<Config> = {
+  maxRequestRatePerSecond: parseIntOptional(__ENV.RPS),
+  mainStageDurationMinutes: parseIntOptional(
+    __ENV.MAIN_TEST_STAGE_DURATION_MINS,
+  ),
+  cooldownStageDurationMinutes: parseIntOptional(
+    __ENV.MAIN_TEST_STAGE_DURATION_MINS,
+  ),
+};
+
+export default function rampingRequestRateProfile({
+  scenario = 'ramping_rate',
+  ...defaultConfig
+}: Config & { scenario?: string }): Options {
+  const {
+    mainStageDurationMinutes,
+    maxRequestRatePerSecond,
+    cooldownStageDurationMinutes,
+  } = mergeObjects(defaultConfig, overrides);
```

Comment:
Could we use lodash's `merge` util here instead of our owwn?

Reply:
I wasn't too sure on the advice around pulling in too many other libraries into K6, but I'll read up on it and happily pull some in to make life easier and cleaner if they don't advise against it.

Reply:
Seems that so long as you don't include any libraries without too many upstream dependencies, it's OK to do so, so have brought this in thanks.  Also some libs don't play nice with K6 given its Go implementation, so worth just bearing this in mind when pulling more in.